src/cursor-doc/undo.ts

```diff
@@ -114,17 +114,21 @@ export class UndoManager<T> {
   undo(c: T) {
     if (this.undos.length) {
       const step = this.undos.pop();
-      step.undo(c);
-      this.redos.push(step);
+      if (step) {
+        step.undo(c);
+        this.redos.push(step);
+      }
```

Comment:
More TypeScript silliness. The length check means pop will always work but TypeScript can't figure that out.